### ISSUE TITLE
Assertion failure when INSERT with after row triggers with ORCA

### ISSUE THREAD

**Issue by l-wang** [greenplum-db/gpdb#7410]
### Greenplum version or build
5X_STABLE

### Step to reproduce the behavior
```
postgres=# CREATE TEMP TABLE PKTABLE (ptest1 int PRIMARY KEY) distributed by (ptest1);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "pktable_pkey" for table "pktable"
CREATE TABLE
postgres=# INSERT INTO PKTABLE VALUES(42);
INSERT 0 1
postgres=# CREATE TEMP TABLE FKTABLE (ftest1 int8 REFERENCES PKTABLE(ptest1)) distributed by (ftest1);
WARNING:  Referential integrity (FOREIGN KEY) constraints are not supported in Greenplum Database, will not be enforced.
CREATE TABLE
postgres=# EXPLAIN INSERT INTO FKTABLE VALUES(42);
                            QUERY PLAN
------------------------------------------------------------------
 Insert  (cost=0.00..0.02 rows=1 width=8)
   ->  Result  (cost=0.00..0.00 rows=1 width=12)
         ->  Result  (cost=0.00..0.00 rows=1 width=8)
               ->  Result  (cost=0.00..0.00 rows=1 width=8)
                     ->  Result  (cost=0.00..0.00 rows=1 width=1)
 Optimizer status: PQO version 3.32.4
(6 rows)

postgres=# INSERT INTO FKTABLE VALUES(42);
NOTICE:  Releasing segworker groups to finish aborting the transaction.
WARNING:  Any temporary tables for this session have been dropped because the gang was disconnected (session id = 49)
ERROR:  Unexpected internal error (execMain.c:3327)  (seg1 10.64.5.130:25433 pid=68114) (cdbdisp.c:254)
DETAIL:  FailedAssertion("!(planGen == PLANGEN_PLANNER)", File: "execMain.c", Line: 3327)
ERROR:  could not connect to segment: initialization of segworker group failed (cdbgang.c:237)
```

The above test is part of the `alter_table` test, however, this never fails on CI, so it seems that the INSERT always falls back to the legacy planner on CI.

**Comment by d** [comment]
I talked to Alexandra offline Wednesday. Posting our findings here for posterity.

The syscache access function `trigger_enabled` is broken during the 5.0 development cycle when `pg_trigger.tgenabled` changed type from `bool` to `char`. c.f. https://github.com/greenplum-db/gpdb/blob/cfcf035c47e318e0ddc691ea7fe5ed1dfc3dece9/src/backend/utils/cache/lsyscache.c#L1660

In the versions of Postgres that Greenplum 5 and 6 are based on, `bool` is simply a typedef to an 8-bit integer. But the C++ code will treat `bool` as a _real_ boolean and hence give it boolean semantics -- one of those is that the only boolean values are 1 and 0. That means that 

I am quite guilty of not fixing this because I knew about this 2 years ago when I noticed `alter_table` failing `installcheck` on macOS, but I was distracted by something "more broken" and promptly forgot about this.

I submitted a patch at #7442 .
